Kickstarter-iOS/Features/RootTabBar/RootTabBarViewController.swift

```diff
@@ -260,46 +260,70 @@ public final class RootTabBarViewController: UITabBarController, MessageBannerVi
-
-        if let imageData = try? Data(contentsOf: imageUrl) {
-          let (defaultImage, selectedImage) = tabbarAvatarImageFromData(imageData)
-          _ = self.tabBarItem(atIndex: index)
-            ?|> profileTabBarItemStyle(isLoggedIn: true, isMember: data.isMember)
-            ?|> UITabBarItem.lens.image .~ defaultImage
-            ?|> UITabBarItem.lens.selectedImage .~ selectedImage
-        } else {
-          let sessionConfig = URLSessionConfiguration.default
-          let session = URLSession(configuration: sessionConfig, delegate: nil, delegateQueue: .main)
-          let dataTask = session.dataTask(with: avatarUrl) { [weak self] avatarData, _, _ in
-            guard let avatarData = avatarData else { return }
-            try? avatarData.write(to: imageUrl, options: [.atomic])
-
-            let (defaultImage, selectedImage) = tabbarAvatarImageFromData(avatarData)
-            _ = self?.tabBarItem(atIndex: index)
-              ?|> profileTabBarItemStyle(isLoggedIn: true, isMember: data.isMember)
-              ?|> UITabBarItem.lens.image .~ defaultImage
-              ?|> UITabBarItem.lens.selectedImage .~ selectedImage
-          }
-          dataTask.resume()
-        }
+        setProfileImage(with: data, avatarUrl: avatarUrl, index: index)
+      }
+    }
+  }
+
+  fileprivate func setProfileImage(with data: TabBarItemsData, avatarUrl: URL?, index: Int) {
+    guard
+      data.isLoggedIn == true,
+      let avatarUrl = avatarUrl,
+      let dir = NSSearchPathForDirectoriesInDomains(.cachesDirectory, .userDomainMask, true).first
+    else { return }
+
+    let hash = avatarUrl.absoluteString.hashValue
+    let imagePath = "\(dir)/tabbar-avatar-image-\(hash).dat"
+    let imageUrl = URL(fileURLWithPath: imagePath)
+
+    if let imageData = try? Data(contentsOf: imageUrl) {
+      let (defaultImage, selectedImage) = tabbarAvatarImageFromData(imageData)
+      _ = self.tabBarItem(atIndex: index)
+        ?|> profileTabBarItemStyle(isLoggedIn: true, isMember: data.isMember)
+        ?|> UITabBarItem.lens.image .~ defaultImage
+        ?|> UITabBarItem.lens.selectedImage .~ selectedImage
+    } else {
+      let sessionConfig = URLSessionConfiguration.default
+      let session = URLSession(configuration: sessionConfig, delegate: nil, delegateQueue: .main)
+      let dataTask = session.dataTask(with: avatarUrl) { [weak self] avatarData, _, _ in
+        guard let avatarData = avatarData else { return }
+        try? avatarData.write(to: imageUrl, options: [.atomic])
+
+        let (defaultImage, selectedImage) = tabbarAvatarImageFromData(avatarData)
+        _ = self?.tabBarItem(atIndex: index)
+          ?|> profileTabBarItemStyle(isLoggedIn: true, isMember: data.isMember)
+          ?|> UITabBarItem.lens.image .~ defaultImage
+          ?|> UITabBarItem.lens.selectedImage .~ selectedImage
       }
+      dataTask.resume()
     }
   }
 
+  fileprivate func isDashboardViewControllerDisplayable() -> Bool {
```

Comment:
This function just checks if the `DashboardTabViewController` is available to display.